system-contracts/bootloader/bootloader.yul

```diff
@@ -3651,14 +3622,14 @@ object "Bootloader" {
 
             // Need to prevent the compiler from optimizing out similar operations, 
             // which may have different meaning for the offline debugging 
-            function unoptimized(val) -> ret {
+            function $llvm_NoInline_llvm$_unoptimized(val) -> ret {
                 ret := add(val, callvalue())
             }
 
             /// @notice Triggers a VM hook. 
             /// The server will recognize it and output corresponding logs.
             function setHook(hook) {
-                mstore(VM_HOOK_PTR(), unoptimized(hook))
+                mstore(VM_HOOK_PTR(), $llvm_NoInline_llvm$_unoptimized(hook))
```

Comment:
Is it used to prevent the optimization of memory write?

Reply:
yes

Reply:
I don't understand why some manipulation with the value helps. I mean let's assume that llvm can optimize `mstore` because it knows the value by this offset is not read anywhere, what will change if the value does not compile time known?
The only case I can imagine is when you are writing the same value as was by this offset, then it can help - but is it a case here?

Reply:
So far as I can see, it does not optimize because it is never read, it rather optimizes because it can detect that the written value is the same in the places near to each other. If you believe it is more appropriate to mark the parent function `setHook` as unoptimized, I'll do that

Reply:
I don't believe it is more appropriate to mark the parent function. I just trying to understand what exactly caused optimization, because looks a little weird from the compiler side